cabal-install/Distribution/Client/Init.hs

```diff
@@ -209,15 +238,25 @@ getLicense flags = do
   lic <-     return (flagToMaybe $ license flags)
          ?>> fmap (fmap (either UnknownLicense id))
                   (maybePrompt flags
-                    (promptList "Please choose a license" listedLicenses (Just BSD3) display True))
+                    (promptList "Please choose a license" listedLicenses
+                     (Just BSD3) displayLicense True))
 
-  if isLicenseInvalid lic
-    then putStrLn promptInvalidOtherLicenseMsg >> getLicense flags
-    else return $ flags { license = maybeToFlag lic }
+  case checkLicenseInvalid lic of
+    Just msg -> putStrLn msg >> getLicense flags
+    Nothing  -> return $ flags { license = maybeToFlag lic }
 
   where
-    isLicenseInvalid (Just (UnknownLicense t)) = any (not . isAlphaNum) t
-    isLicenseInvalid _ = False
+    displayLicense l | needSpdx  = prettyShow (licenseToSPDX l)
+                     | otherwise = display l
+
+    checkLicenseInvalid (Just (UnknownLicense t))
```

Comment:
Not required now, but do you think we could change the type of `InitFlags.license` to `Either License SPDX.License`?

Reply:
@23Skidoo I tried that briefly but in the interest to get this implemented faster I went for the hackier approach and wanted to postpone this to a future PR; doing it properly would require quite a bit of refactoring in multiple places